src/lib_ccx/hardsubx_decoder.c

```diff
@@ -544,6 +544,352 @@ void hardsubx_process_frames_linear(struct lib_hardsubx_ctx *ctx, struct encoder
 	activity_progress(100, cur_sec / 60, cur_sec % 60);
 }
 
+void process_hardsubx_linear_frames_and_normal_subs(struct lib_hardsubx_ctx *hard_ctx, struct encoder_ctx *enc_ctx, struct lib_ccx_ctx *ctx)
```

Comment:
Does this function just have a lot of copy pasted code?

Reply:
Yeah, not exactly duplication but moving it to that method.

That piece of code has been used several places with little or no modification. including the `else` branch of the `!ctx->multiprogram` and since I needed the functionality in this feature as well, I figured that moving the code to the function will reduce some duplication overall and I will call the function appropriately in all the places where the code shows up. 

The code is already duplicated in the existing codebase, as a side effect of working on this issue I hope to generalize the function and call it appropriately.

Reply:
@cfsmp3